instrumentation/opentelemetry-instrumentation-pymssql/src/opentelemetry/instrumentation/pymssql/__init__.py

```diff
@@ -0,0 +1,184 @@
+from opentelemetry.instrumentation.pymssql.version import __version__
+
+_DATABASE_SYSTEM = "mssql"
+
+
+class _PyMSSQLConnectMethodArgsTuple(NamedTuple):
+    server: str | None = None
+    user: str | None = None
+    password: str | None = None
+    database: str | None = None
+    timeout: int | None = None
+    login_timeout: int | None = None
+    charset: str | None = None
+    as_dict: bool | None = None
+    host: str | None = None
+    appname: str | None = None
+    port: str | None = None
+    conn_properties: str | None = None
+    autocommit: bool | None = None
+    tds_version: str | None = None
+
+
+class _PyMSSQLDatabaseApiIntegration(dbapi.DatabaseApiIntegration):
+    def wrapped_connection(
+        self,
+        connect_method: Callable[..., Any],
+        args: tuple[Any, Any],
+        kwargs: dict[Any, Any],
+    ):
+        """Add object proxy to connection object."""
+        connection = connect_method(*args, **kwargs)
+        connect_method_args = _PyMSSQLConnectMethodArgsTuple(*args)
+
+        self.name = self.database_system
+        self.database = kwargs.get("database") or connect_method_args.database
+
+        user = kwargs.get("user") or connect_method_args.user
+        if user is not None:
+            self.span_attributes["db.user"] = user
+
+        port = kwargs.get("port") or connect_method_args.port
+        host = kwargs.get("server") or connect_method_args.server
+        if host is None:
+            host = kwargs.get("host") or connect_method_args.host
+            if host is not None:
+                # The host string can include the port, separated by either a coma or
+                # a column
+                for sep in (":", ","):
+                    if sep in host:
+                        tokens = host.rsplit(sep)
+                        host = tokens[0]
+                        if len(tokens) > 1:
+                            port = tokens[1]
+        if host is not None:
+            self.span_attributes["net.peer.name"] = host
+        if port is not None:
+            self.span_attributes["net.peer.port"] = port
+
+        charset = kwargs.get("charset") or connect_method_args.charset
+        if charset is not None:
+            self.span_attributes["db.charset"] = charset
+
+        tds_version = (
+            kwargs.get("tds_version") or connect_method_args.tds_version
+        )
+        if tds_version is not None:
+            self.span_attributes["db.protocol.tds.version"] = tds_version
+
+        return dbapi.get_traced_connection_proxy(connection, self)
+
+
+class PyMSSQLInstrumentor(BaseInstrumentor):
+    def instrumentation_dependencies(self) -> Collection[str]:
+        return _instruments
+
+    def _instrument(self, **kwargs):
+        """Integrate with the pymssql library.
+        https://github.com/pymssql/pymssql/
+        """
+        tracer_provider = kwargs.get("tracer_provider")
+
+        dbapi.wrap_connect(
```

Comment:
Is [sqlcommenting](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/37f85bf8cc235b807557c8a1aff9fa3b52b3d111/instrumentation/opentelemetry-instrumentation-psycopg2/src/opentelemetry/instrumentation/psycopg2/__init__.py#L21-L33) (example) going to be supported?

Reply:
I'd like to at least add an initial version without support for this feature.

Reply:
Fair! I've created a new issue for that here, feel free to pick up later: https://github.com/open-telemetry/opentelemetry-python-contrib/issues/3199